File: report-generator/src/main/java/com/teamscale/report/util/CommandLineLogger.java
package com.teamscale.report.util;

/** Logger that prints all output to the console. */
public class CommandLineLogger implements ILogger {

	@Override
	public void debug(String message) {
		System.out.println(message);
	}

	@Override
	public void info(String message) {
		System.out.println(message);
	}

	@Override
	public void warn(String message) {
		System.err.println(message);
	}

	@Override
	public void warn(String message, Throwable throwable) {
		System.err.println(message);
		if (throwable != null) {
			throwable.printStackTrace();
		}
	}

	@Override
	public void error(Throwable throwable) {
		throwable.printStackTrace();
	}

	@Override
	public void error(String message, Throwable throwable) {
		System.err.println(message);
		if (throwable != null) {
			throwable.printStackTrace();
		}
	}
}

File: report-generator/src/main/kotlin/com/teamscale/report/util/CommandLineLogger.kt
package com.teamscale.report.util

/** Logger that prints all output to the console.  */
class CommandLineLogger : ILogger {
	override fun debug(message: String) {
		println(message)
	}

	override fun info(message: String) {
		println(message)
	}

	override fun warn(message: String) {
		System.err.println(message)
	}

	override fun warn(message: String, throwable: Throwable?) {
		System.err.println(message)
		throwable?.printStackTrace()
	}

	override fun error(throwable: Throwable) {
		throwable.printStackTrace()
	}

	override fun error(message: String, throwable: Throwable?) {
		System.err.println(message)
		throwable?.printStackTrace()
	}
}
